### PR TITLE
[kueuectl] Clearing cluster queues after each creation on "Creating a ClusterQueue" tests.

### LINKED ISSUE #2454: Flaky integration test: Should print cluster queues list with paging
**What happened**:
Integration test failed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2452/pull-kueue-test-integration-main/1803655079367020544.

**What you expected to happen**:
No flake

**How to reproduce it (as minimally and precisely as possible)**:
Repeat the build.

**Anything else we need to know?**:

```
{Expected
    <string>: "...AGE
    cluster..."
to equal               |
    <string>: "...AGE
    cq1    ..." failed [FAILED] Expected
    <string>: "...AGE
    cluster..."
to equal               |
    <string>: "...AGE
    cq1    ..."
In [It] at: /home/prow/go/src/sigs.k8s.io/kueue/test/integration/kueuectl/list_test.go:178 @ 06/20/24 05:11:36.562
}
```

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
We need to clear cluster queues after each test cases on "Creating a ClusterQueue" tests to fix "Should print cluster queues list with paging" integration test and don't show this cluster queues on the list.

#### Which issue(s) this PR fixes:

Fixes #2454 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/kueuectl/create_test.go
Comment (trasc) [inline test/integration/kueuectl/create_test.go:1]: I'd prefer to have something more symmetrical, maybe we can do it with  [`ginkgo.DeferCleanup` ](https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup)

Comment (mbobrovskyi) [reply test/integration/kueuectl/create_test.go:1]: Looks like `ginkgo.DeferCleanup` not allow to use it on `When` block.
```
You must call DeferCleanup inside a setup node (e.g. BeforeEach, BeforeSuite,
  AfterAll...) or a subject node (i.e. It).  You can't call DeferCleanup at the
  top-level or in a container node - use the After* family of setup nodes
  instead.
```

**Review by trasc — COMMENTED** [review]
/lgtm

**Pull request merged**